nym-api/nym-api-requests/src/ecash/models.rs

```diff
@@ -272,19 +297,167 @@ pub struct AggregatedExpirationDateSignatureResponse {
 
     #[schemars(with = "String")]
     #[serde(with = "crate::helpers::date_serde")]
+    #[schema(value_type = String, example = "1970-01-01")]
     pub expiration_date: Date,
 
     #[schemars(with = "PlaceholderJsonSchemaImpl")]
+    #[schema(value_type = Vec<openapi_schema::AnnotatedExpirationDateSignature>)]
     pub signatures: Vec<AnnotatedExpirationDateSignature>,
 }
 
 #[derive(Serialize, Deserialize, JsonSchema, ToSchema)]
 pub struct AggregatedCoinIndicesSignatureResponse {
     pub epoch_id: u64,
     #[schemars(with = "PlaceholderJsonSchemaImpl")]
+    #[schema(value_type = openapi_schema::Signature)]
     pub signatures: Vec<AnnotatedCoinIndexSignature>,
 }
 
+/// duplicate types from `nym-compact-ecash`, but these do derive `ToSchema```
+pub mod openapi_schema {
+    #![allow(dead_code)]
+    use nym_compact_ecash::common_types::{G2Projective, Scalar};
+
+    use super::*;
+
+    #[derive(ToSchema)]
+    pub struct AnnotatedExpirationDateSignature {
+        pub signature: Signature,
+        pub expiration_timestamp: u32,
+        pub spending_timestamp: u32,
+    }
+
+    #[derive(ToSchema)]
+    pub struct AnnotatedCoinIndexSignature {
+        pub signature: Signature,
+        pub index: u64,
+    }
+
+    #[derive(ToSchema)]
+    pub struct Signature {
+        #[schema(value_type = String)]
+        pub(crate) h: G1Projective,
+        #[schema(value_type = String)]
+        pub(crate) s: G1Projective,
+    }
+
+    #[derive(ToSchema)]
+    pub struct PublicKeyUser {
+        #[schema(value_type = String)]
+        pub(crate) pk: G1Projective,
+    }
+
+    #[derive(ToSchema)]
+    pub struct BlindedSignature {
+        #[schema(value_type = String)]
+        pub h: G1Projective,
+        #[schema(value_type = String)]
+        pub c: G1Projective,
+    }
+
+    #[derive(ToSchema)]
+    pub struct VerificationKeyAuth {
+        #[schema(value_type = String)]
+        pub(crate) alpha: G2Projective,
+        #[schema(value_type = Vec<String>)]
+        pub(crate) beta_g1: Vec<G1Projective>,
+        #[schema(value_type = Vec<String>)]
+        pub(crate) beta_g2: Vec<G2Projective>,
+    }
+
+    #[derive(ToSchema)]
+    pub struct WithdrawalRequest {
+        #[schema(value_type = String)]
+        joined_commitment_hash: G1Projective,
+        #[schema(value_type = String)]
+        joined_commitment: G1Projective,
+        #[schema(value_type = Vec<String>)]
+        private_attributes_commitments: Vec<G1Projective>,
+        zk_proof: WithdrawalReqProof,
+    }
+
+    #[derive(ToSchema)]
+    pub struct WithdrawalReqProof {
+        #[schema(value_type = [u64; 4], format = Binary)]
```

Comment:
I think `Scalar` is also using the same serialisation format as G1 and G2 points, i.e. it's hex-encoded. Please double check it!

Reply:
yep
```rust
impl fmt::Debug for Scalar {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let tmp = self.to_bytes();
        write!(f, "0x")?;
        for &b in tmp.iter().rev() {
            write!(f, "{:02x}", b)?;
        }
        Ok(())
    }
}
```

added `content_encoding = "Base16"`

Reply:
But that's a debug implementation which has nothing to do with serde. I was asking about how it's serialised

Reply:
there is no serde impl in the docs, but a chapter on [serialization](https://docs.rs/bls12_381/latest/bls12_381/notes/serialization/index.html) mentions that  `G_1` group uses `Scalar`s (elements of `F_p` )  for its coordinates